dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -390,28 +390,34 @@ private void deleteSelfFromLambdaEvents(User user) {
         lambdaEventDAO.findByUser(user).stream().forEach(lambdaEvent -> lambdaEvent.setUser(null));
     }
 
-    @DELETE
+    @PATCH
     @Timed
     @UnitOfWork
-    @Path("/user/{userId}")
+    @Path("/user/{userId}/bannedStatus")
     @RolesAllowed("admin")
-    @Operation(operationId = "terminateUsers", description = "Terminate user if possible.", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
-    @ApiResponse(responseCode = HttpStatus.SC_OK + "", description = "Successfully deleted user", content = @Content(schema = @Schema(implementation = Boolean.class)))
+    @Operation(operationId = "banUser", description = "Update banned status of user if possible. Removes all tokens for banned users.", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    @ApiResponse(responseCode = HttpStatus.SC_OK + "", description = "Successfully banned user", content = @Content(schema = @Schema(implementation = User.class)))
     @ApiResponse(responseCode = HttpStatus.SC_FORBIDDEN + "", description = HttpStatusMessageConstants.FORBIDDEN)
     @ApiResponse(responseCode = HttpStatus.SC_NOT_FOUND + "", description = USER_NOT_FOUND_DESCRIPTION)
-    @ApiOperation(value = "Terminate user if possible.", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = Boolean.class, nickname = "terminateUser")
-    public boolean terminateUser(
-        @ApiParam(hidden = true) @Parameter(hidden = true, name = "user")@Auth User authUser,  @ApiParam("User to terminate") @PathParam("userId") long targetUserId) {
-        // note this terminates the user but leaves behind a tombstone to prevent re-login
+    @ApiResponse(responseCode = HttpStatus.SC_BAD_REQUEST + "", description = "This user is already banned/unbanned.")
+    @ApiOperation(value = "Updated banned status of user if possible. Removes all tokens for banned users.", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = User.class, nickname = "banUser")
+    public User banUser(
+        @ApiParam(hidden = true) @Parameter(hidden = true, name = "user")@Auth User authUser,  @ApiParam("User to terminate") @PathParam("userId") long targetUserId,
+        @ApiParam("isBanned") @Parameter(name = "isBanned") boolean isBanned) {
+        // note this bans the user but leaves behind a tombstone to prevent re-login
         checkUser(authUser, authUser.getId());
 
         User targetUser = userDAO.findById(targetUserId);
         checkUserExists(targetUser);
 
-        invalidateTokensForUser(targetUser);
-
-        targetUser.setBanned(true);
-        return true;
+        if (isBanned == targetUser.isBanned()) {
```

Comment:
I lean towards not throwing an error for this. Make it idempotent, and if it's the same as before, just proceed.